tools/swift-format/Tests/SwiftFormatTests/OperatorWhitspaceTests.swift

```diff
@@ -0,0 +1,41 @@
+import SwiftSyntax
+import XCTest
+
+@testable import Rules
+
+public class OperatorWhitespaceTests: DiagnosingTestCase {
+  public func testInvalidOperatorWhitespace() {
+    XCTAssertFormatting(
+      OperatorWhitespace.self,
+      input: """
```

Comment:
Can you add a couple more tests?

* `for number in 1 ... 5 {}`: This should remove the spaces.
* `for number in -10 ... -5 {}`: This is harder. We can't just remove the spaces, or the compiler will greedily treat `...-` as the operator. We can't just remove the space between `-10` and `...` because then it would be parsed as a postfix operator instead of an infix operator. Since we don't want to produce wrong code by reformatting, we only have two options here:
   1. Don't remove the spaces at all.
   2. Remove the spaces but surround the right-hand expression with parentheses.

The second option is more correct according to our style guide, but also a bit more invasive than the first, because it requires us to rewrite expressions instead of just token whitespace.

In either case, we need to know when it's safe to remove the spaces after a range operator. Well, it's safe when the characters in the token immediately following it aren't also characters that Swift would parse as part of an operator, right? At first I thought we'd have to duplicate the entire character set that Swift supports, which would be a massive table, but then it occurred to me, `-10 ... -5` is parsed like this:

    [prefixOperator("-"), integerLiteral("10"), spacedBinaryOperator("..."), prefixOperator("-"), integerLiteral("5")]

So, I think we can accurately say that if we see a range operator as a `spacedBinaryOperator` **and** the next token is a `prefixOperator`, we **shouldn't** remove the spaces. (Let's do the easy thing for now, and if we want to explore paren-wrapping later, we can do it as a follow up.)

@harlanhaskins Does this seem correct?